### PR TITLE
Feature/vendor tool

### PR DESCRIPTION
**Pull request by andydotxyz**
### Description:
Move tools/modvendor to "fyne vendor" command.
This means that others can use this to package up the fyne dependencies (tested with fyne-io/examples, for instance).

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: cmd/fyne/vendor.go [hunk 1]
```diff
@@ -19,92 +9,72 @@ import (
...
 	// goModFile is the go module file
 	goModFile = "go.mod"
 	// goModVendorFile is the vendor module file
 	goModVendorFile = "vendor/modules.txt"
-	// glwfMod is the glfw module
-	glwfMod = "github.com/go-gl/glfw/v3.3/glfw"
-	// glwfModSrcDir is the glfw dir containing the c source code to copy
-	glwfModSrcDir = "glfw"
+	// glfwModNew is the glfw module regexp for 3.3+
+	glfwModNew = "github.com/go-gl/glfw/(v(\\d+\\.)?(\\*|\\d)?(\\*|\\d+))/glfw"
```
Comment (lucor) [inline cmd/fyne/vendor.go]:
```suggestion
	glfwModNew = `github\.com/go-gl/glfw/(v(\d+\.)?(\*|\d)?(\*|\d+))/glfw`
```
alternatively if we won't be strict about version match:

`glfwModNew = "github.com/go-gl/glfw/(v[^/]+)/glfw"`

File: cmd/fyne/vendor.go [hunk 2]
```diff
@@ -19,92 +9,72 @@ import (
...
 		if len(s) != 3 {
 			continue
 		}
-		if s[1] != module {
-			continue
+
+		moduleVersion := ""
+		moduleTarget := ""
+		r, _ := regexp.Compile(glfwModNew)
+		if r.Match([]byte(s[1])) {
```
Comment (lucor) [inline cmd/fyne/vendor.go]: Apply early exit in this block ?

**Comment by andydotxyz** [comment]
Thanks for the feedback, I will apply the changes right after merge

File: cmd/fyne/vendor.go (hunk 1 above)
Comment (andydotxyz) [reply cmd/fyne/vendor.go]: Nice hint on the string formatting - that had escaped me :)

**Pull request merged**